packages/s3-cache/src/index.ts

```diff
@@ -79,12 +127,15 @@ export class S3Cache implements CacheStrategy {
         })
       nextContinuationToken = token
 
-      for (const { Key: key } of contents) {
-        if (!key) continue
-        if (key.endsWith('.json') || key.endsWith('.html')) {
-          await this.client.deleteObject({ Bucket: this.bucketName, Key: key })
-        }
-      }
+      const keyToDelete = contents.reduce<string[]>(
```

Comment:
I believe we still can optimize it, listObjects also returns up to 1k keys, but not all those keys are going to match to what we need to delete, we should separate that:

1. go through all the objects and collect all keys what we need to delete
2. send deleteObjects request to each 1k of keys

Reply:
done